File: scrapers/torrentgalaxy.py
"""TorrentGalaxy scraper - HTML scraping."""

from urllib.parse import quote
from scrapers.base import BaseScraper
from scrapers import client, utils


class Source(BaseScraper):
    name = 'TorrentGalaxy'
    base_url = 'https://tgx.rs'

    def search(self, query):
        url = f'{self.base_url}/torrents.php?search={quote(query)}&sort=seeders&order=desc'
        soup = client.get_soup(url)
        if not soup:
            return []
        results = []
        for row in soup.select('div.tgxtablerow'):
            # Name
            name_tag = row.select_one('a.txlight')
            if not name_tag:
                # Alternative selector
                name_tag = row.select_one('div.tgxtablecell a[href*="/torrent/"]')
            name = utils.clean_name(name_tag.get_text()) if name_tag else ''
            if not name:
                continue
            # Magnet
            magnet_tag = row.select_one('a[href^="magnet:"]')
            if not magnet_tag:
                continue
            magnet = magnet_tag['href']
            # Size - look for the cell containing size info
            cells = row.select('div.tgxtablecell')
            size_text = ''
            seeders = 0
            for cell in cells:
                text = cell.get_text(strip=True)
                if any(u in text.upper() for u in ('GB', 'MB', 'KB', 'TB')):
                    size_text = text
                # Seeders are in a span with specific font color or class
                seed_span = cell.select_one('span[title="Seeders/Leechers"] b, font[color="green"]')
                if seed_span:
                    seeders = utils.parse_seeders(seed_span.get_text())
            if not size_text:
                size_text = 'N/A'
            results.append({
                'name': name,
                'size': size_text,
                'size_gb': utils.parse_size(size_text),
                'seeders': seeders,
                'magnet': magnet,
                'provider': self.name,
            })
        return results

File: netlify/functions/lib/scrapers/torrentgalaxy.js
/**
 * TorrentGalaxy scraper - HTML scraping.
 */
const client = require('../client');
const { parseSeeders, parseSize, cleanName } = require('../utils');

const NAME = 'TorrentGalaxy';
const BASE = 'https://tgx.rs';

async function search(query) {
  const url = `${BASE}/torrents.php?search=${encodeURIComponent(query)}&sort=seeders&order=desc`;
  const $ = await client.getSoup(url);
  if (!$) return [];

  const results = [];
  $('div.tgxtablerow').each((_, row) => {
    const $row = $(row);
    const nameTag = $row.find('a.txlight').first();
    const name = cleanName(nameTag.text() || $row.find('a[href*="/torrent/"]').first().text());
    if (!name) return;

    const magnetTag = $row.find('a[href^="magnet:"]').first();
    if (!magnetTag.length) return;
    const magnet = magnetTag.attr('href');

    let sizeText = '';
    let seeders = 0;
    $row.find('div.tgxtablecell').each((_, cell) => {
      const $cell = $(cell);
      const text = $cell.text().trim();
      if (/\d+\s*(GB|MB|KB|TB)/i.test(text)) sizeText = text;
      const seedSpan = $cell.find('font[color="green"], span b').first();
      if (seedSpan.length) {
        const s = parseSeeders(seedSpan.text());
        if (s > 0) seeders = s;
      }
    });

    results.push({
      name,
      size: sizeText || 'N/A',
      size_gb: parseSize(sizeText),
      seeders,
      magnet,
      provider: NAME,
    });
  });
  return results;
}

module.exports = { name: NAME, base_url: BASE, search };
